packages/apollo-codegen-swift/src/language.ts

```diff
@@ -55,38 +43,263 @@ const reservedKeywords = new Set(['associatedtype', 'class', 'deinit', 'enum', '
+          // If this keyword comes after a '.' make sure it's also a reservedMemberKeyword.
+          if (
+            offset == 0 ||
+            fullString[offset - 1] !== "." ||
+            reservedMemberKeywords.has(match)
+          ) {
+            return `\`${match}\``;
+          }
+        }
+        return match;
+      })
+    );
+  }
 
-export function escapeIdentifierIfNeeded(identifier: string) {
-  if (reservedKeywords.has(identifier)) {
-    return "`" + identifier + "`";
+  /**
+   * Escapes the input if it begins with a reserved keyword not valid in member position.
+   *
+   * Most keywords are valid in member position (e.g. after a period), but a few aren't. This
+   * method escapes just those keywords not valid in member position, and therefore must only be
+   * used on input that is guaranteed to come after a dot.
+   * @param input The input containing identifiers to escape.
+   * @returns The input with relevant identifiers escaped.
+   */
+  static memberName(input: string): SwiftSource {
+    return new SwiftSource(
+      // This behaves nearly identically to `SwiftSource.identifier` except for the logic around
+      // offset zero, but it's structured a bit differently to optimize for the fact that most
+      // matched identifiers are at offset zero.
+      input.replace(/[a-zA-Z_][a-zA-Z0-9_]*/g, (match, offset, fullString) => {
+        if (!reservedMemberKeywords.has(match)) {
+          // If we're not at offset 0 and not after a period, check the full set.
+          if (
+            offset == 0 ||
+            fullString[offset - 1] === "." ||
+            !reservedKeywords.has(match)
+          ) {
+            return match;
+          }
+        }
+        return `\`${match}\``;
+      })
+    );
+  }
+
+  /**
+   * Template tag for producing a `SwiftSource` value without performing escaping.
+   *
+   * This is identical to evaluating the template without the tag and passing the result to `new
+   * SwiftSource(…)`.
+   */
+  static raw(
+    literals: TemplateStringsArray,
+    ...placeholders: any[]
+  ): SwiftSource {
+    // We can't just evaluate the original template directly, but we can replicate its semantics.
+    // NB: The semantics of untagged template literals matches String.prototype.concat rather than
+    // the + operator. Since String.prototype.concat is documented as slower than the + operator,
+    // we'll just use individual template strings to do the concatenation.
+    var result = literals[0];
+    placeholders.forEach((elt, i) => {
+      result += `${elt}${literals[i + 1]}`;
+    });
+    return new SwiftSource(result);
+  }
+
+  toString(): string {
+    return this.source;
+  }
+
+  /**
+   * Concatenates multiple `SwiftSource`s together.
+   */
+  concat(...sources: SwiftSource[]): SwiftSource {
+    // Documentation says + is faster than String.concat, so let's use that
+    var result = this.source;
+    for (let elt of sources) {
```

Comment:
silly question: What does `elt` stand for?

Reply:
element.

Reply:
Ah - is that a standard Typescript thing? I obviously wasn't familiar with it and probably would have just used `element` personally, but if it's idiomatic typescript it should probably stay as is.

Reply:
I'm not a TypeScript programmer normally, I actually don't know what the standard conventions are. `elt` is a fairly common language-agnostic shorthand for "element", but I don't know if it's idiomatic to use abbreviated variable names or not in TypeScript code.

Reply:
FWIW I mostly do Swift / Obj-C / Java / Kotlin, and I'd personally never seen that abbreviation before. It'd probably be good to expand to `element` for clarity for others who aren't typically typescript programmers (which unfortunately is the majority of folks working with this part of the codebase). However if you'd prefer to leave it, it's at least clear enough from what it does what the intention is.